src/simple_http.rs

```diff
@@ -73,29 +75,50 @@ impl SimpleHttpTransport {
     }
 
     fn request<R>(&self, req: impl serde::Serialize) -> Result<R, Error>
+    where
+        R: for<'a> serde::de::Deserialize<'a>,
+    {
+        match self.try_request(req) {
+            Ok(response) => Ok(response),
+            Err(err) => {
+                *self.sock.lock().unwrap() = None;
+                Err(err)
+            }
+        }
+    }
+
+    fn try_request<R>(&self, req: impl serde::Serialize) -> Result<R, Error>
```

Comment:
We should wrap this in another function which intercepts every error and assigns `*sock = None` before returning.

Reply:
I think we're already doing this; `request()` wraps `try_request()`.

Reply:
Oh, derp, yes